From: Jon Trowbridge <trow@novell.com>

Enforce minimum maxSleepCount 1, we shouldn't just fail right away if
lockWaitTimeout < LOCK_POLL_INTERVAL.

Index: Store/Lock.cs
===================================================================
--- Store/Lock.cs	(revision 4506)
+++ Store/Lock.cs	(working copy)
@@ -68,6 +68,7 @@
 			bool locked = Obtain();
 			int maxSleepCount = (int) (lockWaitTimeout / LOCK_POLL_INTERVAL);
 			int sleepCount = 0;
+			maxSleepCount = System.Math.Min (maxSleepCount, 1);
 			while (!locked)
 			{
 				if (sleepCount++ == maxSleepCount)
@@ -144,4 +145,4 @@
 			}
 		}
 	}
-}
\ No newline at end of file
+}
